sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -1,43 +1,457 @@
+
+        studentTask.set('DueDate', date.getTime()/1000);
+        studentTask.set('TaskStatus', 're-assigned');
+        studentTask.save({
+            callback: function() {
+                taskRater.close();
+                Ext.toast('Student task successfully updated.');
+                me.rateStudentTask(studentTask);
+            }
+        });
+    },
+
+    unAssignStudentTask: function(studentTask) {
+        var me = this;
+
+        studentTask.erase({
+            callback: function(opts, success) {
+                if (success) {
+                    Ext.toast('Student task unassigned.');
+                    me.getTaskRater().close();
+                }
+            }
+        });
+    },
+
+    saveStudentTask: function(studentTask, callback, scope) {
+        studentTask.save({
+            success: function(rec) {
+                Ext.callback(callback, scope, [rec]);
+            }
+        });
+    },
+
+    saveTask: function() {
+        var me = this,
+            form = me.getTaskEditorForm(),
+            skillsField = me.getSkillsField(),
+            attachmentsField = me.getAttachmentsField(),
+            assignmentsField = me.getAssignmentsField(),
+            courseSection = me.getCourseSelector().getSelection(),
+            record = form.updateRecord().getRecord(),
+            errors;
+
+        record.set('Skills', skillsField.getSkills(false)); // returnRecords
+        record.set('Attachments', attachmentsField.getAttachments(false)); // returnRecords
+        record.set('Assignees', assignmentsField.getAssignees(false)); // returnRecords
+        record.set('CourseSectionID', courseSection.getId());
+
+        errors = record.validate();
+
+        if (errors.length) {
+            Ext.each(errors.items, function(item) {
+                var itemField = form.down('[name='+item.field +']');
+
+                if (itemField) {
+                    itemField.markInvalid(item.message);
+                }
+            });
+            return;
+        }
+        record.save({
+            success: function() {
+                me.getTaskEditor().close();
+                Ext.toast('Task succesfully saved!');
+            }
+        });
+    },
+
+    editStudentTask: function(studentTask) {
+        var me = this,
+            grid = me.getTasksGrid(),
+            taskEditor = me.getTaskEditor(),
+            task = grid.getRowsStore().getById(studentTask.get('TaskID'));
+
+        if (!task || !studentTask) { // is this likely?
+            Ext.Msg.error('Error', 'Unable to find task. Please refresh the page and try again.'); // is this optimal?
+            return;
+        }
+
+        taskEditor.setTask(task);
+        taskEditor.setStudentTask(studentTask);
+        taskEditor.show();
+    },
+
+    editTask: function(taskRecord) {
+        var me = this,
+            taskEditor = me.getTaskEditor();
+
+        if (!taskRecord) {
+            taskRecord = Ext.create('Slate.cbl.model.Task');
```

Comment:
@nbey this is an implicit dependency on the model, avoid calling `Ext.create` or `Ext.widget` within a controller.

Put at the top of the controller:
```
models: [
    'Task@Slate.cbl.model'
]
```

and then you can do: `me.getTaskModel().create({ ...})`